zmon_worker_extras/check_plugins/exasol.py

```diff
@@ -4,89 +4,81 @@
+        self._exasol_pass = conf['exasol_pass']
+        self._exasol_driver = conf['exasol_driver']
 
     def create(self, factory_ctx):
         """
         Automatically called to create the check function's object
         :param factory_ctx: (dict) names available for Function instantiation
         :return: an object that implements a check function
         """
-        return propartial(ExaplusWrapper, cluster=self._exacrm_cluster, password=self._exacrm_pass,
-                          user=self._exacrm_user)
+        return propartial(ExaplusWrapper, cluster=factory_ctx['cluster'], password=self._exasol_pass,
+                          user=self._exasol_user, schema=factory_ctx['schema'],
+                          driver=self._exasol_driver)
 
 
 class ExaplusWrapper(object):
-    def __init__(self, cluster, user='ZALANDO_NAGIOS', password='', schema=False):
+    def __init__(self, cluster, user='', password='', schema='', driver=''):
         self._err = None
         self._out = None
         self.user = user
         self.__password = password
         self.cluster = cluster
         self.schema = schema
-        self.java_opts = ['-Djava.net.preferIPv4Stack=true', '-Djava.awt.headless=true', '-Xmx512m', '-Xms128m']
-        self.exaplus_opts = [
-            '-recoverConnection',
-            'OFF',
-            '-retry',
-            '0',
-            '-lang',
-            'EN',
-            '-q',
-            '-x',
-            '-Q',
-            '10',
-            '-pipe',
-        ]
-        self.jar_file = '/server/exasol/exaplus/current/exaplus.jar'
+        self.driver = driver
 
     def query(self, query):
-        fd, name = tempfile.mkstemp(suffix='.sql', text=True)
         try:
-            fh = os.fdopen(fd, 'w')
-            fh.write('%s\n' % query)
-            fh.flush()
-
-            cmd = ['/usr/bin/java']
-            cmd.extend(self.java_opts)
-            cmd.extend(['-jar', self.jar_file])
-            cmd.extend(['-c', self.cluster])
-            cmd.extend(['-u', self.user])
-            cmd.extend(['-p', self.__password])
-            cmd.extend(self.exaplus_opts)
-            if self.schema:
-                cmd.extend(['-s', self.schema])
-            cmd.extend(['-f', name])
-            # print "EXAPLUS="+" ".join(cmd)
-            sub = subprocess.Popen(cmd, shell=False, stdout=subprocess.PIPE, stderr=subprocess.PIPE, close_fds=True)
-            d_out, d_err = sub.communicate()
-            self._out = d_out
-            self._err = d_err
-        finally:
-            os.unlink(name)
+            connection = connect(
+                driver=self.driver,
+                EXAHOST=self.cluster,
+                EXASCHEMA=self.schema,
+                EXAUID=self.user,
+                EXAPWD=self.__password)
+            cursor = connection.cursor()
+
+            cursor.execute(query)
+            self._out = cursor
+
+        except Exception as err:
+            self._err = err
```

Comment:
I suggest dropping the return ``tuple`` interface. If there is an exception then we either let it raise and bubble or we catch (filter exceptions?) and raise ``CheckError`` with a more informative message. We only return the result in the end.